src/agent/onefuzz-supervisor/src/scheduler.rs

```diff
@@ -131,9 +131,15 @@ impl State<SettingUp> {
     pub async fn finish(self, runner: &mut dyn ISetupRunner) -> Result<SetupDone> {
         let work_set = self.ctx.work_set;
 
-        let reboot = runner.run(&work_set).await?;
+        let output = runner.run(&work_set).await?;
 
-        let done = if reboot {
+        if let Some(output) = output {
+            if !output.exit_status.success {
+                anyhow::bail!("setup script failed: {:?}", output);
```

Comment:
Should we really be throwing an exception here? I think this should be encoded as a specific state like SetupDone::ScriptFailed.
IMHO throwing an exception is supposed to indicate that something went wrong with this code which is not the case.

Reply:
This just preserves the existing behavior; this is the first in a series of PRs to change that behavior. For example, we add new detail to the `Done` state in a forthcoming PR, and then respond to it in the service event handler.